lib/engines/content_object_store/app/views/content_object_store/content_block/editions/schedule_publishing.html.erb

```diff
@@ -4,15 +4,26 @@
 <% content_for :back_link do %>
   <%= render "govuk_publishing_components/components/back_link", {
     href: content_object_store.edit_content_object_store_content_block_edition_path(
-      id: @content_block_edition.id,
+      id: @content_block_document.latest_edition.id,
       content_block_edition: @edition_params,
       step: ContentObjectStore::ContentBlock::EditionsController::EDIT_FORM_STEPS[:review_links],
     ),
   } %>
 <% end %>
 
+<%
+  year_param = @edition_params["scheduled_publication(1i)"]
+  month_param = @edition_params["scheduled_publication(2i)"]
+  day_param = @edition_params["scheduled_publication(3i)"]
+  hour_param = @edition_params["scheduled_publication(4i)"]
+  minute_param =  @edition_params["scheduled_publication(5i)"]
+  is_scheduled_param = params["schedule_publishing"]
+%>
```

Comment:
Is there a more elegant way to do this? Too much logic in the view makes me a bit squiffy

Reply:
thought it would be ok as we aren't actually doing any logic, just destructuring some params to make it more clear what they are, because of ruby date magic... I got the idea from Whitehall's scheduling component, which does a mix of this and inline digging, which seems less desirable https://github.com/alphagov/whitehall/blob/096d3b551f033a7bcb732dbe905914f889f33067/app/views/admin/editions/_scheduled_publication_fields.html.erb 

what would you recommend, constructing an object in the controller? or just using `@edition_params["scheduled_publication(1i)"]` inline?

Reply:
It could be an idea to use these params to create a `DateTime` object and sent that to the view instead. The view would be simplified down slightly as it'll just send methods to a single object, rather than knowing how to compose all the different parts. Though it's hard to see how this rails date splitting interface changing.

I don't think Rails gives us a way to nicely create that object without having to unpick each part :( So we'd just have to move this up to the controller and send a `@published_time` back down. 

I don't have a strong opinion here given the params aren't going to change. I might still spend 10 minutes to see how the idea felt.

Reply:
I'm going to merge as is as this is now blocking work and already such a mega PR, and there's precedent in the Whitehall code, if it is a big issue happy to pick up in a refactoring PR